edgexfoundry/device-sdk-go
# feat: implement support for ProvisionWatchersDir

add 'ProvisionWatchersDir' configuration to support adding provision watchers during device service startup.

closes #1100

Signed-off-by: Chris Hung <chris@iotechsys.com>







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

1.  Run `device-simple` from this branch
2. Verify `Simple-Provision-Watcher` is added to Metadata during startup:
```shell
$ curl localhost:59881/api/v2/provisionwatcher/name/Simple-Provision-Watcher | jq '.'

{
  "apiVersion": "v2",
  "statusCode": 200,
  "provisionWatcher": {
    "created": 1673235954536,
    "modified": 1673235954536,
    "id": "aec1f082-02d1-4019-8c64-66505c6d2873",
    "name": "Simple-Provision-Watcher",
    "labels": [
      "simple"
    ],
    "identifiers": {
      "Address": "simple[0-9]+",
      "Port": "3[0-9]{2}"
    },
    "blockingIdentifiers": {
      "Port": [
        "397",
        "398",
        "399"
      ]
    },
    "profileName": "Simple-Device",
    "serviceName": "device-simple",
    "adminState": "UNLOCKED",
    "autoEvents": [
      {
        "interval": "15s",
        "onChange": false,
        "sourceName": "SwitchButton"
      }
    ]
  }
}

```

## New Dependency Instructions (If applicable)
